### PR TITLE
perf(Async): Add ConfigureAwait(false) where possible

### PR DESCRIPTION
**Pull request by rozele**

### CODE REVIEW

File: ReactWindows/ReactNative/Bridge/JavaScriptBundleLoader.cs [hunk 1]
```diff
@@ -86,11 +86,12 @@ public override string SourceUrl
             {
                 try
                 {
-                    var storageFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(SourceUrl));
-                    using (var stream = await storageFile.OpenStreamForReadAsync())
+                    var storageFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(SourceUrl)).AsTask().ConfigureAwait(false);
+                    System.Diagnostics.Debug.WriteLine(storageFile.Path);
+                    using (var stream = await storageFile.OpenStreamForReadAsync().ConfigureAwait(false))
```
Comment (rozele) [inline ReactWindows/ReactNative/Bridge/JavaScriptBundleLoader.cs]: Delete debug line.

File: ReactWindows/ReactNative/Bridge/ReactInstance.cs [hunk 2]
```diff
@@ -122,7 +122,8 @@ public void Initialize()
                     _bundleLoader.LoadScript(bridge);
 
                     return bridge;
-                });
+                })
+                .ConfigureAwait(false);
```
Comment (rozele) [inline ReactWindows/ReactNative/Bridge/ReactInstance.cs]: delete newline.

File: ReactWindows/ReactNative/Bridge/Queue/MessageQueueThread.cs [hunk 3]
```diff
@@ -309,7 +309,8 @@ public AnyBackgroundMessageQueueThread(string name, Action<Exception> handler)
                     {
                         _handler(ex);
                     }
-                });
+                })
+                .ConfigureAwait(false);
```
Comment (rozele) [inline ReactWindows/ReactNative/Bridge/Queue/MessageQueueThread.cs]: delete newline.

File: ReactWindows/ReactNative/DevSupport/DevServerHelper.cs [hunk 4]
```diff
@@ -180,17 +180,21 @@ public IDisposable StartPollingOnChangeEndpoint(Action onServerContentChanged)
...
+                        using (var response = await onChangePollingClient.GetAsync(onChangeUrl, disposable.Token).ConfigureAwait(false))
                         {
                             if (response.StatusCode == HttpStatusCode.ResetContent)
                             {
                                 DispatcherHelpers.RunOnDispatcher(new DispatchedHandler(onServerContentChanged));
                             }
                         }
                     }
+                    catch (OperationCanceledException ex)
+                    when (ex.CancellationToken == disposable.Token)
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DevServerHelper.cs]: Confirm that this is called when polling ends (or move to different PR)

File: ReactWindows/ReactNative/DevSupport/DisabledDevSupportManager.cs [hunk 5]
```diff
@@ -68,7 +68,8 @@ public string JavaScriptBundleUrlForRemoteDebugging
             await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.High, () =>
             {
                 ExceptionDispatchInfo.Capture(exception).Throw();
-            });
+            })
+            .AsTask().ConfigureAwait(false);
```
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/DisabledDevSupportManager.cs]: Delete newline.

File: ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs [hunk 6]
```diff
@@ -1,6 +1,7 @@
 ﻿using Newtonsoft.Json.Linq;
 using ReactNative.Bridge;
 using System.Reactive.Linq;
+using System.Reactive.Threading.Tasks;
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs]: Make sure this is needed.

File: ReactWindows/ReactNative/Modules/Network/NetworkingModule.cs [hunk 7]
```diff
@@ -201,16 +201,17 @@ public override void OnReactInstanceDispose()
...
             request.Content = new HttpStreamContent(inputStream);
             await ProcessRequestAsync(
                 requestId,
                 useIncrementalUpdates,
                 timeout,
                 request,
                 responseType,
-                token);
+                token)
+                .ConfigureAwait(false);
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Network/NetworkingModule.cs]: Delete newline.

File: ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs [hunk 8]
```diff
@@ -13,7 +14,7 @@ class AsyncStorageModule : NativeModuleBase
         private const string DirectoryName = "AsyncStorage\\";
         private const string FileExtension = ".data";
 
-        private readonly object _gate = new object();
+        private readonly SemaphoreSlim _mutex = new SemaphoreSlim(1, 1);
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs]: >SemaphoreSlim [](start = 25, length = 13)

This change is too substantial, put in separate review.

File: ReactWindows/ReactNative/Views/View/ReactViewManager.cs [hunk 9]
```diff
@@ -1,6 +1,7 @@
 ﻿using ReactNative.Reflection;
 using ReactNative.UIManager;
 using ReactNative.UIManager.Annotations;
+using Windows.Foundation;
 using Windows.UI.Xaml.Controls;
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/View/ReactViewManager.cs]: Clean up changes in this file.

File: ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs [hunk 10]
```diff
@@ -196,7 +196,7 @@ public void Dispose()
             if (!_isDisposed)
             {
                 _messageWriter.WriteString(message);
-                await _messageWriter.StoreAsync();
+                await _messageWriter.StoreAsync().AsTask().ConfigureAwait(false);
                 // TODO: check result of `StoreAsync()`
```
Comment (mpodwysocki) [inline ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs]: what does this mean?  Does the result mean something for failure or success?

**Comment by mpodwysocki** [comment]
:shipit:

File: ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs (hunk 10 above)
Comment (rozele) [inline ReactWindows/ReactNative/DevSupport/WebSocketJavaScriptExecutor.cs]: We don't care that much, it's just debug code anyway ;-)
